js/src/ScatterGL.ts

```diff
@@ -0,0 +1,913 @@
+
+                has_selection : {type: "b", value: false},
+                has_hover     : {type: "b", value: false},
+
+                has_selected_fill    : {type: "b", value: false},
+                has_selected_stroke    : {type: "b", value: false},
+                has_selected_opacity    : {type: "b", value: false},
+                has_unselected_fill  : {type: "b", value: false},
+                has_unselected_stroke  : {type: "b", value: false},
+                has_unselected_opacity  : {type: "b", value: false},
+
+                has_hovered_fill     : {type: "b", value: false},
+                has_hovered_stroke     : {type: "b", value: false},
+                has_hovered_opacity     : {type: "b", value: false},
+                has_unhovered_fill   : {type: "b", value: false},
+                has_unhovered_stroke   : {type: "b", value: false},
+                has_unhovered_opacity   : {type: "b", value: false},
+
+                selected_fill      : { type: "4f", value: [1., 0., 0., 1.0] },
+                selected_stroke    : { type: "4f", value: [1., 0., 0., 1.0] },
+                selected_opacity   : {value: 1.0},
+                unselected_fill    : { type: "4f", value: [1., 0., 0., 1.0] },
+                unselected_stroke  : { type: "4f", value: [1., 0., 0., 1.0] },
+                unselected_opacity : {value: 1.0},
+                hovered_fill       : { type: "4f", value: [0., 1., 0., 1.0] },
+                hovered_opacity    : {value: 1.0},
+                hovered_stroke    : { type: "4f", value: [0., 1., 0., 1.0] },
+                unhovered_fill  : { type: "4f", value: [0., 1., 0., 1.0] },
+                unhovered_stroke  : { type: "4f", value: [0., 1., 0., 1.0] },
+                unhovered_opacity    : {value: 1.0},
+
+                stroke_color_default: {type: "4f", value: [0, 0, 0, 0]},
+
+                colormap: { type: 't', value: null },
+                fill: {type: 'b', value: true },
+                stroke_width: {type: 'f', value: 1.5 },
+                marker_scale: {type: 'f', value: 1.0 }
+            },
+            vertexShader: require('raw-loader!../shaders/scatter-vertex.glsl'),
+            fragmentShader: require('raw-loader!../shaders/scatter-fragment.glsl'),
+            transparent: true,
+            depthTest: false,
+            depthWrite: false,
+            // background reading for blending:
+            // https://limnu.com/webgl-blending-youre-probably-wrong/
+            blending: THREE.CustomBlending,
+
+            blendEquation: THREE.AddEquation,
+            blendSrc: THREE.OneFactor,
+            blendDst: THREE.OneMinusSrcAlphaFactor,
+
+            blendEquationAlpha: THREE.AddEquation,
+            blendSrcAlpha: THREE.OneFactor,
+            blendDstAlpha: THREE.OneMinusSrcAlphaFactor,
+        });
+
+        this.buffer_marker_geometry = new THREE.BufferGeometry().fromGeometry(new THREE.PlaneGeometry());
```

Comment:
This should be replaced by `this.buffer_marker_geometry = new THREE.PlaneBufferGeometry();`, not only because it improves readability, but also because it reduces the number of vertices (12 vs 18 vertices), and there is an unused extra `color` attribute in the `PlaneGeometry`.